File: code/100/solution.cpp
#include "../template/bigint.cpp"
#include "../template/china_reminder.cpp"
#include "../template/euler.cpp"
#include "../template/pell.cpp"
#include "../template/prime.cpp"
#include "../template/prime_cnt.cpp"
#include <iostream>/*{{{*/
#include <map>
#include <math.h>
#include <set>
#include <vector>
const long long e2 = 100;
const long long e3 = e2 * 10;
const long long e4 = e3 * 10;
const long long e5 = e4 * 10;
const long long e6 = e4 * e2;
const long long e7 = e6 * 10;
const long long e8 = e7 * 10;
long long mod      = 1e9 + 7;
using namespace std; /*}}}*/

/*add{{{*/
long long base_add(long long a, long long b) {
    a %= mod;
    b %= mod;
    return (a + b) % mod;
}
long long add(long long a, long long b) { return base_add(a, b); }
long long add(initializer_list<long long> args) {
    long long ret = 0;
    for (auto i : args)
        ret = base_add(ret, i);
    return ret;
}
/*}}}*/

/*{{{mul*/
long long base_mul(long long a, long long b) {
    a %= mod;
    b %= mod;
    long long ret = 0;
    while (b) {
        if (b & 1)
            ret = (ret + a) % mod;
        b >>= 1;
        a = (a + a) % mod;
    }
    return ret;
}
long long mul(long long a, long long b) { return base_mul(a, b); }
long long mul(initializer_list<long long> args) {
    long long ret = 1;
    for (auto i : args)
        ret = base_mul(ret, i);
    return ret;
} /*}}}*/

/*{{{qpow*/
long long qpow(long long a, long long b) {
    long long ret = 1;
    while (b) {
        if (b & 1)
            ret = mul({ret, a});
        a = mul({a, a});
        b >>= 1;
    }
    return ret;
} /*}}}*/

void gen(long long &a, long long &b) {
    auto aa = -3 * a - 4 * b + 3;
    b       = -2 * a - 3 * b + 2;
    a       = aa;
}
int main() {
    long long a = 0, b = 0;
    long long va = 0, vb = 0;
    while (vb < 1e12) {
        gen(a, b);
        if (a < 0) {
            va = -a + 1;
            vb = 1 - b * 2;
        } else {
            va = a;
            vb = b * 2;
        }
        cout << va << ' ' << vb << endl;
    }
    return 0;
}

File: code/100/solution.go
package main /*{{{*/

import (
	"reflect"
	"fmt"
)

const (
	e4 = int64(10000)
	e5 = int64(100000)
	e6 = int64(1000000)
	e7 = int64(10000000)
	e8 = e4*e4
	mod = int64(1e9+7)
) 

func add(list ...interface{}) int64 {/*{{{*/
	if len(list) == 0 {
		return 0
	}
	ret := int64(0)
	for _, i := range list {
		buf := int64(0)
		switch reflect.TypeOf(i).String(){
			case "int":
				buf = int64(i.(int))
			case "int64":
				buf = i.(int64)
		}
		ret = (ret + buf)%mod
	}
	return ret
}/*}}}*/
func mul(list ...interface{}) int64 {/*{{{*/
	if len(list) == 0 {
		return 0
	}
	ret := int64(1)
	for _, i := range list {
		buf := int64(0)
		switch reflect.TypeOf(i).String() {
			case "int":
				buf = int64(i.(int))
			case "int64":
				buf = i.(int64)
		}
		buf %= mod
		base := ret
		ret = 0
		for buf > 0 {
			if (buf&1) == 1 {
				ret = add(ret, base)
			}
			base = add(base, base)
			buf>>=1
		}
	}
	return ret
}/*}}}*/
func qpow(a, b int64) int64 {/*{{{*/
	ret := int64(1)
	for b > 0 {
		if (b&1) == 1 {
			ret = mul(ret, a)
		}
		a = mul(a, a)
		b>>=1
	}
	return ret
}/*}}}*/
/*}}}*/

func gen(a, b int64) (int64, int64) {
	return -3*a -4*b + 3, -2*a -3*b + 2
}
func main() {
	a, b, va, vb := int64(0), int64(0), int64(0), int64(0)
	for vb < int64(1e12) {
		a, b = gen(a, b)
		if a < 0 {
			va = -a+1
			vb = 1-b*2
		} else {
			va = a
			vb = b*2
		}
		fmt.Println(va, vb)
	}
}
